### PR TITLE
Edit schedules for service templates

### PR DESCRIPTION
**Pull request by bdunne**
V2V enhancements allowing schedule edit

https://bugzilla.redhat.com/show_bug.cgi?id=1564255

### CODE REVIEW

**Comment by bdunne** [comment]
@abellotti Please review

**Comment by JPrause** [comment]
@miq-bot add_label blocker

**Comment by JPrause** [comment]
@bdunne if this can be backported can you add the gaprindashvili/yes label.

**Comment by bdunne** [comment]
@miq-bot add_label gaprindashvili/yes

**Comment by JPrause** [comment]
@bdunne were there any other reviews/edits needed? Can this be merged.

**Comment by bdunne** [comment]
Nope, this is ready to go, just needs review.

**Comment by JPrause** [comment]
@bdunne can you nudge someone to review.

**Comment by bdunne** [comment]
@abellotti Can you review when you get a chance?

File: app/controllers/api/subcollections/schedules.rb [hunk 1]
```diff
@@ -2,7 +2,19 @@ module Api
   module Subcollections
     module Schedules
       def schedules_query_resource(object)
-        object ? object.list_schedules : {}
+        object.miq_schedules
```
Comment (abellotti) [inline app/controllers/api/subcollections/schedules.rb]: won't miq_schedules behave differently than list_schedules ?    report vs service template schedules ?

Comment (bdunne) [reply app/controllers/api/subcollections/schedules.rb]: yes, I moved the reporting specific method to the reports controller and removed the `include` for the schedules subcollection

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit fbcc517a537d1c49c4b3928297f14cc35c9fa22d
Author: Alberto Bellotti <abellotti@users.noreply.github.com>
Date:   Wed Jul 25 10:18:24 2018 -0400

    Merge pull request #417 from bdunne/edit_schedules_for_service_templates
    
    Edit schedules for service templates
    (cherry picked from commit 64345d2440cd1504343c7021ac3654fac7bf0a76)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1564255
```

**Pull request merged**